snow/engine/snowman/bootstrap/bootstrapper.go

```diff
@@ -434,11 +436,6 @@ func (b *Bootstrapper) fetch(ctx context.Context, blkID ids.ID) error {
 		return nil
 	}
 
-	// Make sure we don't already have this block
-	if _, err := b.VM.GetBlock(ctx, blkID); err == nil {
-		return b.tryStartExecuting(ctx)
-	}
```

Comment:
Same comment as the other location - we shouldn't be calling `tryStartExecuting` in multiple places per call. Additionally, this can never really be hit anyways... because we only ever try `fetching` a block if we don't have it locally.